packages/block-library/src/table-of-contents/list.tsx

```diff
@@ -12,16 +17,39 @@ const ENTRY_CLASS_NAME = 'wp-block-table-of-contents__entry';
 
 export default function TableOfContentsList( {
 	nestedHeadingList,
+	disableLinkActivation,
+	onClick,
 }: {
 	nestedHeadingList: NestedHeadingData[];
+	disableLinkActivation?: boolean;
+	onClick?: ( event: MouseEvent< HTMLAnchorElement > ) => void;
 } ): WPElement {
+	// Handle onClick callback.
+	const handleOnClick = ( event: MouseEvent< HTMLAnchorElement > ) => {
+		if ( 'function' === typeof onClick ) {
+			onClick( event );
+		}
+	};
+
```

Comment:
@Mamaduka I think so. TypeScript gets annoyed if I try to call an optional callback that might not be defined.

Reply:
@ZebulanStanphill, do you have any suggestions?

Reply:
I also tried to just call `useDispatch` here first but React through an angry error because the hook was not used in the root component. Figured a consumer was the only way.

Reply:
Yeah, can’t use hooks in this component since it’s used in save method as well. No hooks are allowed there.

Reply:
Can't we just do:

```jsx
<a
onClick={ (disableLinkActivation && typeof onClick === 'function') ? onClick : undefined) }
>
```

Reply:
Yep, done.